lib/logcache/client.rb

```diff
@@ -44,8 +44,13 @@ def container_metrics(source_guid:, start_time:, end_time:, envelope_limit: DEFA
 
     private
 
-    def with_request_error_handling(_source_guid)
+    def with_request_error_handling(source_guid)
       tries ||= 3
+      start_time = Time.now
+      time_taken_in_ms = ((Time.now - start_time) * 1000).to_i # convert to milliseconds to get more precise information
```

Comment:
Calculation of `time_taken_in_ms` and `logger.info` must be placed after `yield`.